[Backport 368702 to release 24.11] koboldcpp: 1.79.1 -> 1.81

4evy opened the pull request:
Backports:
#366837
#368702

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

4evy committed:
koboldcpp: update `cudaArches` example

(cherry picked from commit eb0189224b96379b7e3d76c2393f989abed44817)

4evy committed:
koboldcpp: remove `gitUpdater`

(cherry picked from commit 4bd5e1da37c506ac21862d28064977103fb805df)

4evy committed:
koboldcpp: drop `openblas`

Upstream dropped `openblass` since v1.75

(cherry picked from commit 415fe80741e2b2b65c4a4d5f9416d34b20cc9c69)

4evy committed:
koboldcpp: 1.79.1 -> 1.80

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.80
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.79.1...v1.80
(cherry picked from commit 0dae157805a942c99900cc40aa5e907b7c5920dc)

4evy committed:
koboldcpp: add `(darwinMinVersionHook "10.15")`

Bump up minimum version, so no APIs are missing

(cherry picked from commit e62a47c0c8c1f340066165e1d35dab63f519ce6a)

4evy committed:
koboldcpp: disable vulkan for darwin

It's causing build failure

(cherry picked from commit 1ccd253accfb140acbf9da9f56d9dc6d3aece088)

4evy committed:
koboldcpp: 1.80 -> 1.80.1

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.80.1
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.80...v1.80.1
(cherry picked from commit e57e423549cc53355e9f72449bc47501e7f367eb)

4evy committed:
koboldcpp: 1.80.1 -> 1.80.3

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.80.3
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.80.1...v1.80.3
(cherry picked from commit b814f3f6c3d558a647c12248d88f55951e882cab)

DontEatOreo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369092 --extra-nixpkgs-config '{ allowAliases = false; allowBroken = false; allowUnfree = true; checkMeta = true; cudaSupport = true; cudaCapabilities = [ "7.5" ]; cudaArches = [ "sm_75" ]; }'`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

DontEatOreo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369092`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

DontEatOreo commented:
https://gist.github.com/GrahamcOfBorg/6cb2d2c372e4f7c10fd9c8b07e8bcbb0

ofborg is unhappy about Mono? This package doesn't use Mono at all, so it seems like it's a CI/CD issue...

JohnRTitor commented:
It's a target branch issue, unrelated to this PR, but should still be fixed, perhaps we/someone else forgot to backport a particular commit or broke the eval somehow.

JohnRTitor reviewed (commented):
You sure you want to include this commit? https://github.com/NixOS/nixpkgs/pull/369092/commits/d2cddf950c19560a266bf68a41f16a662ff2db7f

According to https://github.com/NixOS/nixpkgs/pull/366837#discussion_r1898890320, darwin sdk on release-24.11 is still 10.12.

DontEatOreo commented:
@JohnRTitor Thanks for catching this! Yea you're right I removed the `revert` for Darwin SDK

DontEatOreo commented:
@ofborg eval

JohnRTitor commented:
@ofborg build koboldcpp

4evy committed:
koboldcpp: 1.80.3 -> 1.81

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.81
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.80.3...v1.81
(cherry picked from commit 55b977f4832ba7354ab66060f4c6e9f07d9be3a2)

DontEatOreo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369092 --extra-nixpkgs-config '{ allowAliases = false; allowBroken = false; allowUnfree = true; checkMeta = true; cudaSupport = true; cudaCapabilities = [ "7.5" ]; cudaArches = [ "sm_75" ]; }'`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

DontEatOreo commented:
https://github.com/NixOS/nixpkgs/actions/runs/12652279354/job/35254578314?pr=369092 can be ignored it's complaining that I didn't backport the treewide change
